apps/namadillo/src/App/Settings/Advanced.tsx

```diff
@@ -1,23 +1,45 @@
 import { ActionButton, Input, Stack } from "@namada/components";
 import SettingsRoute from "App/Settings/routes";
 import { indexerUrlAtom, rpcUrlAtom } from "atoms/settings";
+import { useValidateApiUrl } from "hooks/useValidateApiUrl";
 import { useAtom } from "jotai";
 import { useState } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
 
 export const Advanced = (): JSX.Element => {
   const navigate = useNavigate();
   const location = useLocation();
+  const validateApiUrl = useValidateApiUrl();
   const [currentRpc, setCurrentRpc] = useAtom(rpcUrlAtom);
   const [currentIndexer, setCurrentIndexer] = useAtom(indexerUrlAtom);
   const [rpc, setRpc] = useState(currentRpc);
+  const [rpcError, setRpcError] = useState("");
   const [indexer, setIndexer] = useState(currentIndexer);
+  const [indexerError, setIndexerError] = useState("");
+  const [validatingUrl, setValidatingUrl] = useState(false);
 
-  const onSubmit = (e: React.FormEvent): void => {
+  const onSubmit = async (e: React.FormEvent): Promise<void> => {
     e.preventDefault();
-    setCurrentIndexer(indexer);
-    setCurrentRpc(rpc);
-    navigate(SettingsRoute.index(), { replace: true, state: location.state });
+    setValidatingUrl(true);
```

Comment:
I wonder if we should create query atoms to validate the indexer / rpc urls 🤔 so we don't need to recreate all these flags. wdyt?

Reply:
do you mean like the current implementation of `indexerHeartbeatAtom`?

Reply:
@pedrorezende I moved everything to the mutation solution. Let me know your impressions :)